addons/resource/models/resource.py

```diff
@@ -987,6 +986,129 @@ def _get_unavailable_intervals(self, start, end):
+        if global_leaves:
+            for leave in global_leaves:
+                dt0 = string_to_datetime(leave[0]).astimezone(tz)
+                dt1 = string_to_datetime(leave[1]).astimezone(tz)
+                result.append((max(start, dt0), min(end, dt1), empty_leave))
+        return attendances - Intervals(result)
+
+    def _get_unavailable_intervals_gantt(self, start_datetime, end_datetime):
+        """
+        This method is used in project_enterprise and planning in the gantt view. The goal is to correctly compute the time-offs of an employee.
+        The generic method does not handle the case when an employee has future leaves planned in a calendar that is different than the current calendar linked to the employee.
+        e.a.
+        employee A with calendar C
+        1 personal leave W for the employee A in calendar B
+        1 global leave X in calendar B
+        1 global leave Y in calendar C
+        1 global leave Z with no calendar_id
+        We have to compute the work_intervals for the employee A while taking into account only the relevant leaves (W, Y, Z)
+        :param start_datetime: first day displayed in the gantt view
+        :param end_datetime: last day displayed in the gantt view
+        :return: a dict { res.id : work_intervals } where res.id is the id of the resource, and the work_intervals are all the available working intervals for the resource.
+        """
+        leaves_mapping = defaultdict(dict)
+
+        calendars_set = self.calendar_id
+
+        start_dt = utc.localize(start_datetime) if not start_datetime.tzinfo else start_datetime
+        end_dt = utc.localize(end_datetime) if not end_datetime.tzinfo else end_datetime
+        # fetch the leaves for the current resources
+        personal_leaves_intervals = self._personal_leaves_intervals(start_dt, end_dt)
+        global_leave_per_calendar = self._global_leaves_per_calendar(start_dt, end_dt, calendars_set)
+        dict_attendance_per_calendar = {
+            calendar: calendar._attendance_intervals_batch(start_dt, end_dt, resources=self)
+            for calendar in self.calendar_id
+        }
+        # since the astimezone is a costly operation, we stock the result in a dict in order to avoid the computation in case multiple resources have the same timezone
+        tz_dates = {}
+        for res in self:
+            tz = timezone(res.user_id.tz or res.calendar_id.tz)
+            if (tz, start_dt) in tz_dates:
+                start = tz_dates[(tz, start_dt)]
+            else:
+                start = start_dt.astimezone(tz)
+                tz_dates[(tz, start_dt)] = start
+            if (tz, end_dt) in tz_dates:
+                end = tz_dates[(tz, end_dt)]
+            else:
+                end = end_dt.astimezone(tz)
+                tz_dates[(tz, end_dt)] = end
+
+            personal_leaves = personal_leaves_intervals[res.id] if personal_leaves_intervals.get(res.id) else False
```

Comment:
```suggestion
            personal_leaves = personal_leaves_intervals.get(res.id, False)
```
2nd arg in get is the default/"or_else" value. :slightly_smiling_face: